internal/plugin/connectors/http/generic/external_api.go

```diff
@@ -0,0 +1,75 @@
+package generichttp
+
+import (
+	"fmt"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/http"
+	"gopkg.in/yaml.v2"
+)
+
+// ConfigYAML is an almost-literal representation of the YAML config section
+// of a generic HTTP connector.  It has two purposes:
+//
+//     1.  It allows other internal connectors to configure a generic connector
+//         to which they are delegating their implementation.  That is, it's
+//         a code-level interface equivalent of a .yml file config section.
+//     2.  Technically, it acts as an intermediate form when parsing an actual
+//         .yml file.  That is, we convert from .yml --> ConfigYAML --> config.
+type ConfigYAML struct {
+	CredentialPatterns map[string]string `yaml:"credentialValidation"`
+	Headers map[string]string `yaml:"headers"`
+	ForceSSL bool `yaml:"forceSSL"`
+}
+
+// NewConfigYAML takes the raw cfgBytes and unmarshals them into a ConfigYAML
+// struct.
+func NewConfigYAML(cfgBytes []byte) (*ConfigYAML, error) {
+	cfgYAML := &ConfigYAML{
+		// TODO: Fix CRDs so we can make the default true
+		ForceSSL: false, // Default should be false
+	}
+
+	err := yaml.Unmarshal(cfgBytes, cfgYAML)
+	if err != nil {
+		return nil, err
+	}
+
+	return cfgYAML, nil
+}
+
+// NewNewConnector is a meta-constructor: a function that returns a NewConnector
+// function. If that's confusing, here's an equivalent phrasing: it's a function
+// that creates a connector constructor.  It's intended to be used by other HTTP
+// connectors who delegate their implementation to the generic HTTP connector.
+//
+// It takes a static ConfigYAML -- static in the sense that its represents the
+// connector definition, not user configuration -- which is then merged at
+// runtime with the actual config from the .yml file, allowing ForceSSL to be
+// configured by the user.
+func NewNewConnector(staticCfgYAML *ConfigYAML) (http.NewConnectorFunc, error) {
```

Comment:
Nit: Could we call this NewConnectorConstructor or something? I think NewNew might be a little confusing. It describes what it returns moreso than what it does, I feel.